commit 8f51353d46d9d0b3c040c65fe62fe0a0969dc36d
Author: Julia Lawall <julia.lawall@lip6.fr>
Date:   Tue Jan 17 12:23:21 2017 +0100

    stmmac: add missing of_node_put
    
    
    [ Upstream commit a249708bc2aa1fe3ddf15dfac22bee519d15996b ]
    
    The function stmmac_dt_phy provides several possibilities for initializing
    plat->mdio_node, all of which have the effect of increasing the reference
    count of the assigned value.  This field is not updated elsewhere, so the
    value is live until the end of the lifetime of plat (devm_allocated), just
    after the end of stmmac_remove_config_dt.  Thus, add an of_node_put on
    plat->mdio_node in stmmac_remove_config_dt.  It is possible that the field
    mdio_node is never initialized, but of_node_put is NULL-safe, so it is also
    safe to call of_node_put in that case.
    
    Signed-off-by: Julia Lawall <Julia.Lawall@lip6.fr>
    Acked-by: Alexandre TORGUE <alexandre.torgue@st.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
index ac3d39c..890e4b0 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
@@ -346,6 +346,7 @@ void stmmac_remove_config_dt(struct platform_device *pdev,
 	if (of_phy_is_fixed_link(np))
 		of_phy_deregister_fixed_link(np);
 	of_node_put(plat->phy_node);
+	of_node_put(plat->mdio_node);
 }
 #else
 struct plat_stmmacenet_data *
